reference/common_errors.rst

```diff
@@ -0,0 +1,143 @@
+Commonly Encountered Error Messages
+===================================
+
+.. _error_failed_execute:
+
+Failed to execute task *n*
+--------------------------
+
+This happens when a task fails due to an issue on a particular worker node. For instance, consider a query that generally succeeds but raises an error on one of the workers:
+
+.. code-block:: sql
+
+  CREATE TABLE pageviews (
+    page_id int,
+    good_views int,
+    total_views int
+  );
+
+  INSERT INTO pageviews
+    VALUES (1, 50, 100), (2, 0, 0);
+
+  SELECT create_distributed_table('pageviews', 'page_id');
+
+  SELECT page_id,
+    good_views / total_views AS goodness
+  FROM pageviews;
+
+The SELECT query fails:
+
+::
+
+  ERROR:  failed to execute task 50
+  STATEMENT:  SELECT page_id, good_views/total_views AS goodness FROM pageviews;
+  ERROR:  XX000: failed to execute task 50
+  LOCATION:  MultiRealTimeExecute, multi_real_time_executor.c:255
+
+To find out what's really going on, we have to examine the database logs inside worker nodes. In our case page_id=1 is stored on one worker and page_id=2 on another. The logs for the latter reveal:
+
+::
+
+  ERROR:  division by zero
+  STATEMENT:  COPY (SELECT page_id, (good_views / total_views) AS goodness FROM pageviews_102480 pageviews WHERE true) TO STDOUT
+  WARNING:  division by zero
+  CONTEXT:  while executing command on localhost:5433
+  WARNING:  22012: division by zero
+  LOCATION:  ReportResultError, remote_commands.c:293
+
+That's because ``total_views`` is zero in a row in shard ``pageviews_102480``.
+
+Common real-life causes for worker failures include PgBouncer failure, invalid concatenation of jsonb objects, and typecasting errors.
+
+Relation *foo* is not distributed
+---------------------------------
+
+This is caused by attempting to mix local and distributed tables in the same query. For an example, with workarounds, see :ref:`join_local_dist`.
```

Comment:
Oh you're right. I wrote "mix" because I thought other things like unions of local and distributed tables would fail, but that actually works now through a distributed subplan.